ui/app/AppMain.qml

```diff
@@ -6,12 +6,23 @@ import "../shared"
 import "./AppLayouts"
 
 RowLayout {
-    property var appSettings
-
-    id: rowLayout
+    id: appMain
     Layout.fillHeight: true
     Layout.fillWidth: true
 
+    signal settingsLoaded()
+
+    Connections {
+        target: profileModel
+        onProfileChanged: {
+            appSettings.fileName = utilsModel.join3Paths(utilsModel.getDataDir(), 'qt', profileModel.profile.address)
+            settingsLoaded()
+            if (appSettings.locale !== "en") {
+                profileModel.changeLocale(appSettings.locale)
+            }
+        }
+    }
+
```

Comment:
Why is setting this via a signal needed? Could the `appSettings` component take a `QtProperty` of the current profile filename from nim instead?